astor/code_gen.py

```diff
@@ -700,7 +701,7 @@ def _handle_string_constant(self, node, value, is_joined=False):
 
             def recurse(node):
                 for value in node.values:
-                    if isinstance(value, ast.Str):
+                    if has_ast_str and isinstance(value, ast.Str):
```

Comment:
ast.Str causes warnings starting with python3.12 , and if you don't need to support python 3.6 and 3.7 it can be avoided completely by checking with
```
def is_str_constant(node: ast.AST) -> bool:
    """Return ``True`` if ``node`` represents a plain string constant."""
    return isinstance(node, ast.Constant) and isinstance(node.value, str)
```